### PR TITLE
Feat/rating for `Park`

### PR DESCRIPTION
**Pull request by tercierp**
In this branch, park data has been modified to include the rating, 

The idea behind the implementation is to minimize calculation time, bearing in mind that we are limited by Firestore 

```kotlin
data class Park(
    val pid: String,
    var name: String,
    var location: ParkLocation,
    var imageReference: String,
    var rating: Float = 0f,
    var nbrRating: Int = 0,
    var capacity: Int,
    var occupancy: Int,
    var events: List<String>,
    var votersUIDs: List<String> = emptyList()
)
```
If a `User` add a new rating we update the rating with the following formula:

$$updatedRating = \frac{newRating + nbrRating \cdot rating}{nbrRating+1}$$

### CODE REVIEW

File: app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt
Comment (Alex720p) [inline app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt:143]: Why are they 2 addRating fns and what's the purpose of adding a rating without being linked to a UID?

File: app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt
Comment (Alex720p) [inline app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt:305]: I might be missing something as the whole viewModel is structured like this and I haven't worked on it but:
What happens when the require fails? I haven't seen the exception being handled here or in the viewModel. Is it supposed to be handled in the UI? Might be worth adding it in the description of the function if that's the case

File: app/src/test/java/com/android/streetworkapp/model/park/ParkRepositoryFirestoreTest.kt
Comment (Alex720p) [inline app/src/test/java/com/android/streetworkapp/model/park/ParkRepositoryFirestoreTest.kt:480]: Might be worth adding a test that tests the corner case when `val currentNbrRating = 1` to ensure that this case is handled if someone works on this code later on

**Review by Alex720p — CHANGES_REQUESTED** [review]
Good code! 
I just have 2 questions that are below.
Side note: There is the code from Hotfix/park mvvm in this branch that I haven't reviewed. I guess you needed it to make the changes/forgot to remove it, might be worth merging the other branch first

**Comment by tercierp** [comment]
> Good code! I have one improvement for the tests and have 2 questions that are below Side note: There is the code from Hotfix/park mvvm in this branch that I haven't reviewed. I guess you needed it to make the changes/forgot to remove it, might be worth merging the other branch first

Yep I'm waiting for the first branch to be reviewed than we will merge this code

File: app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt:143]: This is a vestige of my basic UI debugging to see if the mvvm worked, good call I'm deleting it.

File: app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt:305]: When `require` statement fails, it throws an `IllegalArgumentException` with the message `PID_EMPTY`, preconditions failing means that the developper is not using the function correctly, it's hard for me to see a case where it would be useful for the MVVM to "replace" a bad parameter with a good one without the developper being aware that its use is problematic.

File: app/src/test/java/com/android/streetworkapp/model/park/ParkRepositoryFirestoreTest.kt
Comment (tercierp) [reply app/src/test/java/com/android/streetworkapp/model/park/ParkRepositoryFirestoreTest.kt:450]: This test was added:
```kotlin
@Test
  fun deleteRatingWithCurrentNbrRatingOfOneResetsRatingToZero() = runTest {
    // Arrange
    val pid = "123"
    val rating = 4
    val currentRating = 4.0
    val currentNbrRating = 1

    // Mock Firestore interactions
    `when`(db.collection("parks")).thenReturn(collection)
    `when`(collection.document(pid)).thenReturn(documentRef)
    `when`(documentRef.get()).thenReturn(Tasks.forResult(document))
    `when`(document.getDouble("rating")).thenReturn(currentRating)
    `when`(document.getLong("nbrRating")).thenReturn(currentNbrRating.toLong())

    // Expected values after removing the rating
    val newNbrRating = 0
    val newRating = 0.0

    // Act
    parkRepository.deleteRating(pid, rating)

    // Assert
    verify(documentRef).update(mapOf("rating" to newRating, "nbrRating" to newNbrRating))
  }
```

**Comment by tercierp** [comment]
> > Good code! I have one improvement for the tests and have 2 questions that are below Side note: There is the code from Hotfix/park mvvm in this branch that I haven't reviewed. I guess you needed it to make the changes/forgot to remove it, might be worth merging the other branch first
> 
> Yep I'm waiting for the first branch to be reviewed than we will merge this code

Since the previous PR is merged without any changes we can merge this PR when finished

File: app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt:279]: fair enough, wasn't sure if the app crashing was the expected behavior

**Review by Alex720p — APPROVED** [review]

**Pull request merged**
